akka-stream/src/main/scala/akka/stream/impl/Sinks.scala

```diff
@@ -351,7 +356,7 @@ import org.reactivestreams.Subscriber
       require(maxBuffer > 0, "Buffer size must be greater than 0")
 
       var buffer: Buffer[Received[T]] = _
-      var currentRequest: Option[Requested[T]] = None
+      var currentRequests: mutable.Queue[Requested[T]] = mutable.Queue.empty
```

Comment:
Use `Buffer` here as well, with the max size

Reply:
Fixed